source/extensions/filters/common/ratelimit/ratelimit_impl.cc

```diff
@@ -110,9 +110,11 @@ void GrpcClientImpl::onSuccess(
   callbacks_ = nullptr;
 }
 
-void GrpcClientImpl::onFailure(Grpc::Status::GrpcStatus status, const std::string&,
+void GrpcClientImpl::onFailure(Grpc::Status::GrpcStatus status, const std::string& msg,
                                Tracing::Span&) {
   ASSERT(status != Grpc::Status::WellKnownGrpcStatus::Ok);
+  ENVOY_LOG_TO_LOGGER(Logger::Registry::getLog(Logger::Id::filter), warn,
+                      "rate limit fail, status={} msg={}", status, msg);
```

Comment:
This will log spam at high volume if the upstream RL server is having issues which I don't think is what we want. I would switch this to debug. There should already be a failure stat. If not we can add one?

Reply:
there's nothing record right now when  the upstream RL server is having issues, I think 'warn' is suitable in this scenario

Reply:
As I said it's not suitable for log spamming reasons. If there is no other stat, you can also potentially use `ENVOY_LOG_EVERY_POW_2` or similar.

Reply:
switch it to debug